DiverseSelector/base.py

```diff
@@ -136,59 +179,41 @@ def __init__(self,
         #                          f"than {self.enhanced_sampling_weight} of number of selected"
         #                          f"molecules {self.num_selected_mols}.")
 
-    # def enhanced_sampling(self):
-    #     """Enhanced sampling with random sampling or Monte Carlo sampling."""
-    #     pass
+    def cluster(self, clustering_method:str):
+        """
+        Perform the clastering of the dataset
+        :param clustering_method: selected method for clustering (k-means, DBSCAN etc)
+        :return:
+        """
+        pass
 
-    @abstractmethod
-    def select(self):
-        """Add support of clustering and enhanced sampling."""
+    def select(self, num_selected: int, random_seed=None):
+        """
+        Select the subset molecules with optimal diversity.
+        :param num_selected: amount of molecule that need to be selected
+        :param random_seed: int
+        :return:
+        """
+        self.random_seed = 42 if random_seed is None else random_seed
+        # check if number of clusters is less than number of selected molecules
+        if not self.num_clusters > num_selected:
+            raise ValueError("The number of clusters is great than number of selected molecules.")
+        # check if we have valid number of clusters because selecting 1.5 molecules is not practical
+        if int(num_selected / num_clusters) - num_selected / num_clusters != 0:
+            raise ValueError("The number of molecules in each cluster should be an integer.")
         pass
 
-    @property
-    def subset_diversity(self):
-        """Selected subset diversity."""
+    def subset_diversity(self, subset: np.ndarray, div_metric: str):
+        """
+        Calculate diversity of the subset.
+        :param subset: numpy array of indices which make up a subset
+        :param div_metric: metric for calculating diversity of the subset ("Gini", "Entropy" etc.)
+        :return: float #Scale should be discussed (0 to 1, or 0 to 100 or -1 to 1 or anything else)
```

Comment:
I forgot to mention that we are following `numpy` documentation style, which may benefit the future sphinx HTML file generation.

Reply:
My bad! I used the PyCharm standard documenting style. Can rewrite it

Reply:
This is definitely not a wrong thing! It's following the `Google` coding style instead of `numpy`. It's just a choice. Don't worry about this and you can leave it as it is. I will fix it once this PR is merged. Thanks.